src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterEventListenersAndSubscribersPass.php

```diff
@@ -69,23 +67,28 @@ public function process(ContainerBuilder $container)
         }
     }
 
-    private function addTaggedSubscribers(ContainerBuilder $container, array &$listenerRefs)
+    private function addTaggedServices(ContainerBuilder $container): array
     {
+        $listenerTag = $this->tagPrefix.'.event_listener';
         $subscriberTag = $this->tagPrefix.'.event_subscriber';
-        $taggedSubscribers = $this->findAndSortTags($subscriberTag, $container);
+        $listenerRefs = [];
+        $taggedServices = $this->findAndSortTags([$subscriberTag, $listenerTag], $container);
 
         $managerDefs = [];
-        foreach ($taggedSubscribers as $taggedSubscriber) {
-            [$id, $tag] = $taggedSubscriber;
+        foreach ($taggedServices as $taggedSubscriber) {
+            [$tagName, $id, $tag] = $taggedSubscriber;
             $connections = isset($tag['connection']) ? [$tag['connection']] : array_keys($this->connections);
+            if ($listenerTag === $tagName && !isset($tag['event'])) {
+                throw new InvalidArgumentException(sprintf('Doctrine event listener "%s" must specify the "event" attribute.', $id));
+            }
             foreach ($connections as $con) {
                 if (!isset($this->connections[$con])) {
                     throw new RuntimeException(sprintf('The Doctrine connection "%s" referenced in service "%s" does not exist. Available connections names: "%s".', $con, $id, implode('", "', array_keys($this->connections))));
                 }
 
                 if (!isset($managerDefs[$con])) {
                     $managerDef = $parentDef = $this->getEventManagerDef($container, $con);
-                    while ($parentDef instanceof ChildDefinition) {
+                    while (!$parentDef->getClass() && $parentDef instanceof ChildDefinition) {
```

Comment:
that's a bug fix a lower branch

Reply:
https://github.com/symfony/symfony/pull/39990